integration-tests/tests/src/tests/serialization.ts

```diff
@@ -458,34 +265,23 @@ describe("JSON serialization", () => {
 
         it("toJSON returns a circular structure", function (this: TestContext) {
           const serializable = this.playlists[0].toJSON();
-
           // Check that no props are functions on the serializable object.
           expect(Object.values(serializable).some((val) => typeof val === "function")).equals(false);
-
           // Check that linked list is not a Realm entity.
           expect(serializable.related).not.instanceOf(Realm.Collection);
           // But is a plain Array
           expect(Array.isArray(serializable.related)).equals(true);
-
           // Check that the serializable object is the same as the first related object.
           // (this check only makes sense because of our structure)
-          expect(serializable).equals(serializable.related[0]);
+          const relatedObjects = serializable.related as Array<typeof serializable>;
```

Comment:
not sure if using `typeof serializable` is okay here / in general, not too familiar with practices just yet.